src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -222,40 +222,58 @@ private Version getServerVersion() throws IOException {
     NodesInfo info = new NodesInfo.Builder().addCleanApiParameter("version").build();
     JsonObject result = client.execute(info).getJsonObject();
     if (result == null) {
-      LOG.warn("Couldn't get Elasticsearch version, result is null");
+      LOG.warn("Couldn't get Elasticsearch version (result is null); assuming {}", defaultVersion);
       return defaultVersion;
     }
 
     checkForError(result);
 
     JsonObject nodesRoot = result.get("nodes").getAsJsonObject();
     if (nodesRoot == null || nodesRoot.entrySet().size() == 0) {
-      LOG.warn("Couldn't get Elasticsearch version, nodesRoot is null or empty");
+      LOG.warn(
+          "Couldn't get Elasticsearch version (response nodesRoot is null or empty); assuming {}",
+          defaultVersion
+      );
       return defaultVersion;
     }
 
     JsonObject nodeRoot = nodesRoot.entrySet().iterator().next().getValue().getAsJsonObject();
     if (nodeRoot == null) {
-      LOG.warn("Couldn't get Elasticsearch version, nodeRoot is null");
+      LOG.warn(
+          "Couldn't get Elasticsearch version (response nodeRoot is null); assuming {}",
+          defaultVersion
+      );
       return defaultVersion;
     }
 
     String esVersion = nodeRoot.get("version").getAsString();
+    Version version;
     if (esVersion == null) {
-      LOG.warn("Couldn't get Elasticsearch version, version is null");
-      return defaultVersion;
+      version = defaultVersion;
+      LOG.warn(
+          "Couldn't get Elasticsearch version (response version is null); assuming {}",
+          version
+      );
     } else if (esVersion.startsWith("1.")) {
-      return Version.ES_V1;
+      version = Version.ES_V1;
+      log.info("Detected Elasticsearch version is {}", version);
     } else if (esVersion.startsWith("2.")) {
-      return Version.ES_V2;
+      version = Version.ES_V2;
+      log.info("Detected Elasticsearch version is {}", version);
     } else if (esVersion.startsWith("5.")) {
-      return Version.ES_V5;
+      version = Version.ES_V5;
+      log.info("Detected Elasticsearch version is {}", version);
     } else if (esVersion.startsWith("6.")) {
-      return Version.ES_V6;
+      version = Version.ES_V6;
+      log.info("Detected Elasticsearch version is {}", version);
     } else if (esVersion.startsWith("7.")) {
-      return Version.ES_V7;
+      version = Version.ES_V7;
+      log.info("Detected Elasticsearch version is {}", version);
+    } else {
+      version = defaultVersion;
+      log.info("Detected unexpected Elasticsearch version {}, using {}", esVersion, version);
     }
-    return defaultVersion;
+    return version;
```

Comment:
```
if (version != defaultVersion) {
     log.info(...)
}
```

Reply:
Declined after discussing it on the main PR.